Formula/libosinfo.rb

```diff
@@ -46,7 +51,7 @@ def install
       system "meson", *std_meson_args, *flags, ".."
       system "ninja", "install", "-v"
     end
-    (share/"osinfo/.keep").write ""
+    share.install_symlink HOMEBREW_PREFIX/"share/osinfo"
```

Comment:
Not sure this is the right thing to do. `share` is something like `/usr/local/Cellar/libosinfo/1.9.0/share`. This call to `share.install_symlink` then installs a symlink in `share` called `osinfo`.

However, the contents of `share` will be symlinked into `/usr/local/share`. so `/usr/local/share` will end up containing a broken `osinfo` symlink (that points to itself).

Reply:
`osinfo-db` installs files to `/usr/local/Cellar/osinfo-db/20210426/share/osinfo`, which Homebrew links at the directory level:

```
lrwxr-xr-x  1 lina  admin  41 May  6 17:29 /usr/local/share/osinfo -> ../Cellar/osinfo-db/20210426/share/osinfo
```

For `libosinfo` to have access via `/usr/local/Cellar/libosinfo/1.9.0`, the code above creates this symlink:

```
lrwxr-xr-x  1 lina  admin  24 May  6 17:30 /usr/local/Cellar/libosinfo/1.9.0/share/osinfo -> ../../../../share/osinfo
```

Which points to `/usr/local/share/osinfo`.

All of this works but I agree there should be a better solution.

Reply:
Hm, ok, that answers my question below about when the `osinfo` directory is created. This is fine then.